### ISSUE TITLE
Improve large database imports via db access (Postgres)

### ISSUE THREAD

**Issue by programmerq** [gravitational/teleport#26868]
### Expected behavior:

importing a large database dump should have at least comparable performance compared to directly accessing the database.

### Current behavior:

When attempting to do an import of a 15 GiB pg_dump file that takes about 20 minutes when importing it directly to the database backend, it was observed to have only gotten part of the way through the import in an hour before the user canceled the attempt.

This particular case was seen with postgres and a file produced via pg_dump

Bug details:
- Teleport version
- Recreation steps - compare the time required to import 15 GiB of data via db access vs direct.
- Debug logs

**Comment by Tener** [comment]
I did some tests, and generally, I got both backup (via `pg_dump`) and DB restore to run reasonably fast.

Notably, both workloads were heavily affected by running with `--debug` flag. When enabled for the Database service, the Teleport engine for Postgres will log a tremendous amount of information. I'll change this behavior, as this is rarely useful, and the performance impact is severe. 

After disabling the debug logs, the restore (`psql postgres://... -f backup.sql`) ran smoothly with negligible performance impact from Teleport.

The `pg_dump` itself sees some degradation, but the exact cause is yet to be understood. I do have a workaround (don't parse individual server messages, and instead forward them to the client with `io.Copy`), but it would be worthwhile to find a more general solution.

@programmerq If you know the details of the original deployment, it might be helpful here. It is possible that there is yet another inefficiency that needs addressing, which doesn't show up in my tests because of a different setup.
